### PR TITLE
feat: Make suggest next questions configurable

### PR DESCRIPTION
**Pull request by leehuwuj**
Fix: https://github.com/run-llama/create-llama/issues/244

## Summary by CodeRabbit

- **New Features**
	- Introduced a configuration for managing the "next questions" feature, enhancing flexibility in user interactions.
	- Added functionality to suggest follow-up questions based on conversation history, improving user engagement.
	- Implemented a service for processing and storing private files, facilitating efficient data management.

- **Bug Fixes**
	- Improved response generation logic to conditionally yield suggested questions based on user settings.

- **Documentation**
	- Updated project dependencies to include `pydantic-settings` for enhanced configuration management.

### CODE REVIEW

File: helpers/env-variables.ts [hunk 1]
```diff
@@ -487,33 +487,28 @@ It\\'s cute animal.
 };
 
 const getTemplateEnvs = (template?: TemplateType): EnvVar[] => {
-  if (template === "multiagent") {
```
Comment (marcusschiesser) [inline helpers/env-variables.ts:490]: ah good catch!

File: helpers/env-variables.ts [hunk 2]
```diff
@@ -487,33 +487,28 @@ It\\'s cute animal.
...
-      {
-        name: "AGENT_DUMMY_PORT",
-        value: "8004",
-      },
-    ];
-  } else {
-    return [];
+  const nextQuestionEnvs: EnvVar[] = [
+    {
+      name: "NEXT_QUESTION_ENABLE",
```
Comment (marcusschiesser) [inline helpers/env-variables.ts:492]: don't we have the convention for citation system prompt that if it's empty we don't use it? should be consistent

File: templates/components/services/python/file.py
Comment (marcusschiesser) [inline templates/components/services/python/file.py:1]: also remove from templates

File: templates/components/services/python/suggestion.py
Comment (marcusschiesser) [inline templates/components/services/python/suggestion.py:1]: also remove from templates

File: templates/components/services/python/suggestion.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+from app.api.routers.models import Message
+
+logger = logging.getLogger("uvicorn")
+
+
+class NextQuestionSettings(BaseSettings):
+    enable: bool = True
+    prompt_template: str = (
```
Comment (marcusschiesser) [inline templates/components/services/python/suggestion.py:16]: not needed if we have it in env-variables.ts

File: templates/components/services/python/suggestion.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import logging
+from typing import List, Optional
+
+from llama_index.core.prompts import PromptTemplate
+from llama_index.core.settings import Settings
+from pydantic import BaseModel
+from pydantic_settings import BaseSettings, SettingsConfigDict
```
Comment (marcusschiesser) [inline templates/components/services/python/suggestion.py:7]: might be confusing to add pydantic_settings just for one case?

File: templates/components/services/python/suggestion.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            for message in reversed(messages):
+                if message.role == "user":
+                    last_user_message = f"User: {message.content}"
+                elif message.role == "assistant":
+                    last_assistant_message = f"Assistant: {message.content}"
+                if last_user_message and last_assistant_message:
+                    break
+            conversation: str = f"{last_user_message}\n{last_assistant_message}"
+
+            output: NextQuestions = await Settings.llm.astructured_predict(
```
Comment (marcusschiesser) [inline templates/components/services/python/suggestion.py:64]: `Settings.llm.astructured_predict` is not always working - see https://github.com/run-llama/create-llama/issues/244 - how about using Bastians suggestion?

Comment (leehuwuj) [reply templates/components/services/python/suggestion.py]: Makes sense, Python then should have the same behavior with TS.

**Pull request merged**